types/enhanced-resolve/index.d.ts

```diff
@@ -19,31 +18,31 @@ import Resolver = require('./lib/Resolver');
 import Tapable = require('tapable');
 
 declare namespace Resolve {
-    function sync(path: string, request: string): ResolveResult;
-    function sync(context: ResolveContext, path: string, request: string): ResolveResult;
+    function sync(path: string, request: string): string;
+    function sync(context: ResolveContext, path: string, request: string): string;
 
     function context(path: string, request: string, callback: LoggingCallbackWrapper): void;
     function context(context: ResolveContext, path: string, request: string, callback: LoggingCallbackWrapper): void;
 
     namespace context {
-        function sync(path: string, request: string): ResolveResult;
-        function sync(context: ResolveContext, path: string, request: string): ResolveResult;
+        function sync(path: string, request: string): string;
+        function sync(context: ResolveContext, path: string, request: string): string;
     }
 
     function loader(path: string, request: string, callback: LoggingCallbackWrapper): void;
     function loader(context: ResolveContext, path: string, request: string, callback: LoggingCallbackWrapper): void;
 
-    namespace context {
-        function sync(path: string, request: string): ResolveResult;
-        function sync(context: ResolveContext, path: string, request: string): ResolveResult;
+    namespace loader {
+        function sync(path: string, request: string): string;
+        function sync(context: ResolveContext, path: string, request: string): string;
     }
 
-    function create(path: string, request: string, callback: LoggingCallbackWrapper): void;
-    function create(context: ResolveContext, path: string, request: string, callback: LoggingCallbackWrapper): void;
+    //function create(options: ResolverFactory.ResolverOption): (path: string, request: string, callback: LoggingCallbackWrapper) => void;
+    function create(options: ResolverFactory.ResolverOption): (context: ResolveContext, path: string, request: string, callback: LoggingCallbackWrapper) => void;
```

Comment:
Maybe this would work?
```ts
function create(options: ResolverFactory.ResolverOption): {
    (path: string, request: string, callback: LoggingCallbackWrapper): void;
    (context: ResolveContext, path: string, request: string, callback: LoggingCallbackWrapper): void;
}
```